eras/conway/impl/src/Cardano/Ledger/Conway/UTxO.hs

```diff
@@ -1,26 +1,49 @@
 {-# LANGUAGE FlexibleContexts #-}
 {-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE RecordWildCards #-}
 {-# LANGUAGE TypeFamilies #-}
 {-# OPTIONS_GHC -Wno-orphans #-}
 
 module Cardano.Ledger.Conway.UTxO () where
 
 import Cardano.Ledger.Alonzo.UTxO (
-  AlonzoScriptsNeeded,
+  AlonzoScriptsNeeded (..),
   getAlonzoScriptsHashesNeeded,
   getAlonzoScriptsNeeded,
  )
+import Cardano.Ledger.Conway.Core (ConwayEraTxBody (..), EraTxBody (..))
 import Cardano.Ledger.Conway.Era (ConwayEra)
+import Cardano.Ledger.Conway.Governance (VotingProcedure (..))
 import Cardano.Ledger.Conway.TxBody ()
+import Cardano.Ledger.Credential (Credential (..))
 import Cardano.Ledger.Crypto (Crypto)
 import Cardano.Ledger.Mary.UTxO (getConsumedMaryValue)
-import Cardano.Ledger.UTxO (EraUTxO (..))
+import Cardano.Ledger.UTxO (EraUTxO (..), UTxO)
+import Data.Foldable (Foldable (..))
+import Data.Maybe (mapMaybe)
+import Lens.Micro ((^.))
+
+getConwayScriptsNeeded ::
+  ConwayEraTxBody era =>
+  UTxO era ->
+  TxBody era ->
+  AlonzoScriptsNeeded era
+getConwayScriptsNeeded utxo txb = getAlonzoScriptsNeeded utxo txb <> voteScripts
```

Comment:
This requires a `Monoid` instance that was added in `cardano-ledger-alonzo-1.3.1.0`. This means a lower bound need to be updated in  the cabal file